src/core_atmosphere/physics/physics_wrf/module_sf_sfclay.F

```diff
@@ -879,62 +963,77 @@ SUBROUTINE SFCLAY1D(J,UX,VX,T1D,QV1D,P1D,dz8w1d,                &
-      DO 400 I=its,ite
-        IF(XLAND(I)-1.5.GT.0.)THEN                                           
-          HFX(I)=FLHC(I)*(THGB(I)-THX(I))                                
+!!!  390 CONTINUE                                                                 
+!!!      DO 400 I=its,ite
+!$acc loop gang vector
+        DO I=its,ite
+          IF(XLAND(I)-1.5.GT.0.)THEN                                           
+            HFX(I)=FLHC(I)*(THGB(I)-THX(I))                                
 !         IF ( PRESENT(ISFTCFLX) ) THEN
 !            IF ( ISFTCFLX.NE.0 ) THEN
 ! AHW: add dissipative heating term (commented out in 3.6.1)
 !               HFX(I)=HFX(I)+RHOX(I)*USTM(I)*USTM(I)*WSPDI(I)
 !            ENDIF
 !         ENDIF
-        ELSEIF(XLAND(I)-1.5.LT.0.)THEN                                       
-          HFX(I)=FLHC(I)*(THGB(I)-THX(I))                                
-          HFX(I)=AMAX1(HFX(I),-250.)                                       
-        ENDIF                                                                  
-  400 CONTINUE                                                                 
-
-  405 CONTINUE                                                                 
-         
-      DO I=its,ite
-         IF((XLAND(I)-1.5).GE.0)THEN
-           ZL=ZNT(I)
-         ELSE
-           ZL=0.01
-         ENDIF
-         CHS(I)=UST(I)*KARMAN/DENOMQ(I)
-!        GZ2OZ0(I)=ALOG(2./ZNT(I))
-!        PSIM2(I)=-10.*GZ2OZ0(I)
-!        PSIM2(I)=AMAX1(PSIM2(I),-10.)
-!        PSIH2(I)=PSIM2(I)
-         CQS2(I)=UST(I)*KARMAN/DENOMQ2(I)
-         CHS2(I)=UST(I)*KARMAN/DENOMT2(I)
-      ENDDO
-                                                                        
-  410 CONTINUE                                                                   
+          ELSEIF(XLAND(I)-1.5.LT.0.)THEN                                       
+            HFX(I)=FLHC(I)*(THGB(I)-THX(I))                                
+            HFX(I)=AMAX1(HFX(I),-250.)                                       
+          ENDIF                                                                  
+       END DO                                                                 
+!$acc end parallel
+!!!  400 CONTINUE                                                                 
+    END IF
+!!!  405 CONTINUE                                                                 
+!$acc parallel vector_length(32)
+!$acc loop gang vector private(ZL)         
+        DO I=its,ite
+          IF((XLAND(I)-1.5).GE.0)THEN
+            ZL=ZNT(I)
+          ELSE
+            ZL=0.01
+          ENDIF
+          CHS(I)=UST(I)*KARMAN/DENOMQ(I)
+!         GZ2OZ0(I)=ALOG(2./ZNT(I))
+!         PSIM2(I)=-10.*GZ2OZ0(I)
+!         PSIM2(I)=AMAX1(PSIM2(I),-10.)
+!         PSIH2(I)=PSIM2(I)
+          CQS2(I)=UST(I)*KARMAN/DENOMQ2(I)
+          CHS2(I)=UST(I)*KARMAN/DENOMT2(I)
+        ENDDO
+!$acc end parallel
+!-!-!$acc update host(QFX,LH,HFX,CHS,CQS2,CHS2)
```

Comment:
There are a few places with these deactivated (?) `update host` directives. Will these be needed if we eventually intend to make use of variables like `QFX`, `LH`, `HFX`, etc. on the host, either for computation or to write them to model output files?